packages/tokens/fungible/src/extensions/allowlist/storage.rs

```diff
@@ -0,0 +1,212 @@
+    /// * data - `[]`
+    ///
+    /// # Security Warning
+    ///
+    /// **IMPORTANT**: This function bypasses authorization checks and should
+    /// only be used:
+    /// - During contract initialization/construction
+    /// - In admin functions that implement their own authorization logic
+    ///
+    /// Using this function in public-facing methods creates significant
+    /// security risks as it could allow unauthorized allowlist
+    /// modifications.
+    pub fn disallow_user(e: &Env, user: &Address) {
+        // Set the user as not allowed
+        let key = AllowListStorageKey::Allowed(user.clone());
+        e.storage().persistent().set(&key, &false);
+        e.storage().persistent().extend_ttl(&key, BALANCE_TTL_THRESHOLD, BALANCE_EXTEND_AMOUNT);
+
+        // Emit event
+        emit_user_disallowed(e, user);
+    }
+
+    // ################## OVERRIDDEN FUNCTIONS ##################
+
+    /// Transfers `amount` of tokens from `from` to `to`.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to Soroban environment.
+    /// * `from` - The address holding the tokens.
+    /// * `to` - The address receiving the transferred tokens.
+    /// * `amount` - The amount of tokens to be transferred.
+    ///
+    /// # Errors
+    ///
+    /// * [`FungibleTokenError::UserNotAllowed`] - When either `from` or `to` is
+    ///   not allowed.
+    /// * Also refer to [`Base::transfer`] errors.
+    pub fn transfer(e: &Env, from: &Address, to: &Address, amount: i128) {
+        // Check if both addresses are allowed
+        if !AllowList::allowed(e, from) {
+            panic_with_error!(e, FungibleTokenError::UserNotAllowed);
+        }
+        if !AllowList::allowed(e, to) {
+            panic_with_error!(e, FungibleTokenError::UserNotAllowed);
+        }
+
+        // Call the base implementation
+        Base::transfer(e, from, to, amount);
+    }
+
+    /// Transfers `amount` of tokens from `from` to `to` using the
+    /// allowance mechanism. `amount` is then deducted from `spender`s
+    /// allowance.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to Soroban environment.
+    /// * `spender` - The address authorizing the transfer, and having its
+    ///   allowance consumed during the transfer.
+    /// * `from` - The address holding the tokens which will be transferred.
+    /// * `to` - The address receiving the transferred tokens.
+    /// * `amount` - The amount of tokens to be transferred.
+    ///
+    /// # Errors
+    ///
+    /// * [`FungibleTokenError::UserNotAllowed`] - When either `from`, `to`, or
+    ///   `spender` is not allowed.
+    /// * Also refer to [`Base::transfer_from`] errors.
+    pub fn transfer_from(e: &Env, spender: &Address, from: &Address, to: &Address, amount: i128) {
+        // Check if all addresses are allowed
+        if !AllowList::allowed(e, spender) {
+            panic_with_error!(e, FungibleTokenError::UserNotAllowed);
+        }
```

Comment:
this case is an interesting one, do we need to check `spenser`, because technically they don't operate with their own funds

Reply:
yes, just double checked with solidity library, they didn't put such a bound, removing this one